modules/apps/message-boards/message-boards-web/src/main/resources/META-INF/resources/message_boards/edit_message.jsp

```diff
@@ -459,35 +459,46 @@ if (portletTitleBasedNavigation) {
 	</aui:form>
 </clay:container-fluid>
 
-<aui:script require='<%= npmResolvedPackageName + "/message_boards/js/MBPortlet.es as MBPortlet" %>'>
-	new MBPortlet.default({
-		constants: {
-			ACTION_PUBLISH: '<%= WorkflowConstants.ACTION_PUBLISH %>',
-			ACTION_SAVE_DRAFT: '<%= WorkflowConstants.ACTION_SAVE_DRAFT %>',
-			CMD: '<%= Constants.CMD %>',
-		},
-		currentAction:
-			'<%= (message == null) ? Constants.ADD : Constants.UPDATE %>',
-
-		<c:if test="<%= message != null %>">
-			<portlet:resourceURL id="/message_boards/get_attachments" var="getAttachmentsURL">
-				<portlet:param name="messageId" value="<%= String.valueOf(message.getMessageId()) %>" />
-			</portlet:resourceURL>
-
-			getAttachmentsURL: '<%= getAttachmentsURL %>',
-		</c:if>
-
-		namespace: '<portlet:namespace />',
-		rootNode: '#<portlet:namespace />mbEditPageContainer',
-
-		<c:if test="<%= message != null %>">
-			<portlet:renderURL var="viewTrashAttachmentsURL" windowState="<%= LiferayWindowState.POP_UP.toString() %>">
-				<portlet:param name="mvcRenderCommandName" value="/message_boards/view_deleted_message_attachments" />
-				<portlet:param name="redirect" value="<%= currentURL %>" />
-				<portlet:param name="messageId" value="<%= String.valueOf(message.getMessageId()) %>" />
-			</portlet:renderURL>
+<%
+Map<String, Object> taglibContext = HashMapBuilder.<String, Object>put(
+	"constants",
+	HashMapBuilder.<String, Object>put(
+		"ACTION_PUBLISH", WorkflowConstants.ACTION_PUBLISH
+	).put(
+		"ACTION_SAVE_DRAFT", WorkflowConstants.ACTION_SAVE_DRAFT
+	).put(
+		"CMD", Constants.CMD
+	).build()
+).put(
+	"currentAction", (message == null) ? Constants.ADD : Constants.UPDATE
+).put(
+	"messageId", (message != null) ? messageId : null
+).put(
+	"rootNodeId", liferayPortletResponse.getNamespace() + "mbEditPageContainer"
+).build();
+%>
 
-			viewTrashAttachmentsURL: '<%= viewTrashAttachmentsURL %>',
-		</c:if>
-	});
-</aui:script>
\ No newline at end of file
+<c:if test="<%= message != null %>">
+	<portlet:resourceURL id="/message_boards/get_attachments" var="getAttachmentsURL">
+		<portlet:param name="messageId" value="<%= String.valueOf(message.getMessageId()) %>" />
+	</portlet:resourceURL>
+
+	<portlet:renderURL var="viewTrashAttachmentsURL" windowState="<%= LiferayWindowState.POP_UP.toString() %>">
+		<portlet:param name="mvcRenderCommandName" value="/message_boards/view_deleted_message_attachments" />
+		<portlet:param name="redirect" value="<%= currentURL %>" />
+		<portlet:param name="messageId" value="<%= String.valueOf(message.getMessageId()) %>" />
+	</portlet:renderURL>
+
+	<%
+	taglibContext.put("getAttachmentsURL", getAttachmentsURL);
+	taglibContext.put("viewTrashAttachmentsURL", viewTrashAttachmentsURL);
+	%>
+
+</c:if>
+
+<liferay-frontend:component
+	componentId='<%= liferayPortletResponse.getNamespace() + "MBPorlet" %>'
```

Comment:
I think there's a typo here, `MBPorlet` instead of `MBPortlet`.

In any case, is this really needed? Does it have any specific API that we expect someone else to use it via `Liferay.componentReady`?

Unless it's needed and used, I would skip it

Reply:
yeah, there is a typo, thanks for catching up. And not, it is not used by any other component, so we will remove this `componentId` attribute